keps/0001a-meta-kep-implementation.md

```diff
@@ -0,0 +1,213 @@
+---
+
+# Meta KEP Implementation
+
+## Table of Contents
+
+- [Table of Contents](#table-of-contents)
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Why another KEP?](#why-another-kep)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [Implementation Details / Notes / Constraints](#implementation-details--notes--constraints)
+    - [Define](#define)
+    - [Organize](#organize)
+    - [Visibility and Automation](#visibility-and-automation)
+    - [Constraints](#constraints)
+- [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+
+## Summary
+
+Drive KEP adoption through improved process, documentation, visibility, and automation.
+
+## Motivation
+
+The KEP process is the standardized structure for proposing changes to the Kubernetes project.
+
+In order to graduate KEPs to GA, we must iterate over the implementation details.
+
+This KEP seeks to define actionable / delegable items to move the process forward.
+
+Finally, by submitting a KEP, we gain an opportunity to dogfood the process and further identify areas for improvement.
+
+### Why another KEP?
+
+As [KEP-1] is currently de facto for the project, we must be careful to make changes to it in an iterative and atomic fashion.
+
+When proposing a KEP, we action over some unit of work, usually some area of code.
+
+In this instance, we treat [KEP-1] as the unit of work. That said, this would be considered a meta-KEP of the meta-KEP.
+
+### Non-Goals
+
+- API Review process
+- Feature request triage
+- Developer guide
+
+## Proposal
+
+### Implementation Details / Notes / Constraints
+
+#### Define
+
+- Refine existing KEP documentation
+- Define KEP [DACI]
+- Glossary of terms (enhancement, KEP, feature, etc.)
+- KEP Workflow
+  - KEP states
+  - Entry / exit criteria
+
+#### Organize
+
+- Move KEPs from flat-files to a directory structure:
+```
+├── keps # top-level directory
+│   ├── sig-beard # SIG directory
+|   |   ├── 9000-beard-implementation-api # KEP directory
+|   |   |   ├── kep.md (required) # KEP (multi-release work)
+|   |   |   ├── experience_reports (required) # user feedback
+|   |   |   │   ├── alpha-feedback.md
+|   |   |   │   └── beta-feedback.md
+|   |   |   ├── features # units of work that span approximately one release cycle
+|   |   |   │   ├── feature-01.md
+|   |   |   │   ├── ...
+|   |   |   │   └── feature-n.md
+|   |   |   ├── guides
+|   |   |   |   ├── guide-for-developers.md (required)
+|   |   |   |   ├── guide-for-teachers.md (required)
+|   |   |   |   ├── guide-for-operators.md (required)
+|   |   |   |   └── guide-for-project-maintainers.md
+|   |   |   ├── index.json (required) # used for site generation e.g., Hugo
+|   |   |   ├── metadata.yaml (required) # used for automation / project tracking
+|   |   └── └── OWNERS
+│   ├── sig-foo
+|   ├── ...
+|   └── sig-n
+```
+
+metadata.yaml would contain information that was previously in a KEP's YAML front-matter:
+
+```
+---
```

Comment:
This strikes me as a lot of metadata required to write a KEP.

Reply:
should be generated by automation in the general case

Reply:
+1 to automation managing the metadata as much as possible.